tools/scripts/generic.mk

```diff
@@ -6,6 +6,10 @@ export CFLAGS
 export CC
 export AR
 
+# Used by the build utils
+export EXTRA_INC_PATHS
```

Comment:
I don't think that if a variable is exported, its behaviour should change.

Reply:
I did have some troubles with the exports. 
If you export EXTRA_INC_PATHS it will be added to environment and in some rule make is called recursively. Then, for EXTRA_INC_PATHS which is initialized with += will duplicate its values.